services/spar/src/Spar/Intra/Galley.hs

```diff
@@ -91,3 +110,20 @@ isEmailValidationEnabledTeam tid = do
                == Just FeatureStatusEnabled
            )
     )
+
+-- | Update a team member.
+updateTeamMember ::
+  (MonadIO m, HasCallStack, MonadError SparError m, MonadSparToGalley m) =>
+  UserId ->
+  TeamId ->
+  Role ->
+  m ()
+updateTeamMember u tid role = do
```

Comment:
if you're only passing in a role here, you shouldn't construct the entire member from scratch and write that.  you should only update the role.  otherwise you may overwrite current or yet-to-be-introduced attributes of that member with default values.

Reply:
AFAICT this endpoint only exists to update a team member's permissions (equivalent to role). Are you concerned about not yet existing future functionality?

Reply:
> Are you concerned about not yet existing future functionality?

yes.  that, and i didn't actually check the types.  i was expecting there to be something else, too.